sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,462 @@
+            if compare is False: returns the calculations here from Boerner
+
+
+        """
+
+        assert wavelength in self.wavelength_centers, "Choose one integer value from wavelength centers: ' + str(self.wavelength_centers) or 'all'"
+
+        # define variables to call from channel dictionary
+        var = self.property_per_channel_dict(wavelength)
+        Reflectance = var['primary'] * var['secondary']
+        Transmission_efficiency = var['fp_filter'] * var['ent_filter']
+
+
+        eff_area = var['geoarea'] * var['ccd'] * var['contam'] * Transmission_efficiency  * Reflectance
+
+        if compare:
+            self.eff_area = var['effarea']
+
+        else:
+            self.eff_area = eff_area
+        return self.eff_area
+
+        # TODO: Work on units!! not right as of yet - cm**14?!
+        # REFERENCE IDL CODE:
+        # if usephottoelec and usephottodn:-- not sure why?! - conversions?
+        #       units = 12398. / wave * elecperev / elecperdn
+        # else:
+        #       units = 1
+        # eff_area = ((geoarea * ent_filter * fp_filter * primary * secondary * ccd * contam) + cross_area) * units
+
+
+    def instrument_response(self, wavelength = 94, ssw_gain=False, compare = False):
+        """
+
+        Describes the instrument (AIA) response per wavelength by calculating effective area vs wavelength of the strongest emission lines present in the solar feature
+        \
+            R(\lambda)=A_{eff}(\lambda,t)G(\lambda)
+
+        notes:
+        For a given position in the image plane \mathbf{x} and wavelength channel i , the pixel values can be expressed as,
+
+        p_i(\mathbf{x})=\int_0^{\infty}\mathrm{d}\lambda\,\eta_i(\lambda)\int_{pixel\,\mathbf{x}}\mathrm{d}\theta\,I(\lambda,\theta)
+
+        Here, \eta_i(\lambda,t,\mathbf{x}) is the efficiency function of the i^{th} channel
+
+
+        effective area A_{eff}=A_{geo}R_p(\lambda)R_S(\lambda)T_E(\lambda)T_F(\lambda)D(\lambda,t)Q(\lambda)
+        gain of the CCD-camera system, G(\lambda)=(12398/\lambda/3.65)g
+        flat field function F(\mathbf{x})
+
+           Wavelength response functions: calculate the amount of flux per wavelength
+         R_i(\lambda) which is equivalent to \eta_i as expressed above.
+
+
+            # ccd == quantum efficience ==q
+            # calulate gain across wavelength centers - included in master gain of aia image metadata load gain
+            # contam = fluctions of the instruement
+        Parameters
+        ----------
+        :keyword
+        input: string, area from effective area calculation
+
+        input: may need a photon-to-DN unit conversion   # digital number = counts on detectore
+
+        output: outfile of instrument response per channel
+
+        Returns
+        -------
+        :return: float, array describing the response per wavelength of effective area (wavelength response)
+                want units cm^2 DN phot^-1
+        """
```

Comment:
It's probably worth putting a references section in this docstring with a link to the Boerner et al paper.